pkg/persistence/disk_persistence_test.go

```diff
@@ -46,8 +51,30 @@ func TestDiskPersistence_Save(t *testing.T) {
 	}
 }
 
+func TestDiskPersistence_StoragePermission(t *testing.T) {
+	tempDir := "./data_storage"
+
+	err := os.Mkdir(tempDir, 0000) // d---------
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		t.Fatalf("dir [%+v] was supposed to be created", tempDir)
+	}
+	defer os.RemoveAll(tempDir)
+
+	_, err = NewDiskHandle(tempDir)
+	if !strings.Contains(err.Error(), errExpectedRead.Error()) {
+		t.Fatalf("error on read was supposed to be returned")
```

Comment:
Maybe we can follow the convention:
```
t.Errorf(
    "unexpected error\nexpected: %v\nactual:   %v\n",
    expectedError,
    err,
)
```

Reply:
I thought about it, but the full error that is returned would look smth like:
```
cannot read from the storage directory: [open ./data_storage: permission denied]
```
which contains a folder name specific to each operator. `expectedError ` is just a part of the full message that we have control over and I thought that displaying 
```
"unexpected error\nexpected: %v\nactual:   %v\n",
    expectedError,
    err,
```
is not actually true (because we display a partial error message as expected). Idk, it's not biggy and I can change it if you think it adds value.

Reply:
Ok, you're right. We can leave as it is.